### PR TITLE
Plagiarism finder add tests

### PR DESCRIPTION
**Pull request by kipmann**

### CODE REVIEW

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 1]
```diff
@@ -0,0 +1,166 @@
+package de.htwberlin.f4.wikiplag.plagiarism
+
+/**
```
Comment (Tiatzi) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: Scaladoc above class, not imports

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+import scala.util.control.Exception
+
+
+class MatchTest extends AssertionsForJUnit {
+
+  var finder: PlagiarismFinder = _
+  var sc: SparkContext = _
+
+  @Before def setUp() {
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:24]: better create spark context only once and not per test - expensive operation, takes a lot of time

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala (hunk 1 above)
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:3]: Scaladoc above class, not imports

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    //find plagiarisms using default hyper parameters
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    matches.foreach(println)
+    matches.foreach(x => println("[" + (input + input2).substring(x._1.start, x._1.end) + "] Matches: " + x._2))
+    val str2 = matches.toString;
+    if (str2.contains("1993: Goldener Bär der Berliner Filmfestspiele für Das Hochzeitsbankett")) {
+      assert(true)
+    }
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:47]: bool containsExpectedPlagiarismText= str2.contains("1993: Goldener Bär der Berliner Filmfestspiele für Das Hochzeitsbankett")
assert(containsExpectedPlagiarismText)

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    val input2 = raw"Prevailed sincerity behaviour to so do principle mr. As departure at no propriety zealously my. On dear rent if girl view. First on smart there he sense. Earnestly enjoyment her you resources. Brother chamber ten old against. Mr be cottage so related minuter is. Delicate say and blessing ladyship exertion few margaret. Delight herself welcome against smiling its for. Suspected discovery by he affection household of principle perfectly he."
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:54]: str1 obsolete. str2=matches.toString

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+    assertFalse(str2.contains("principle"))
+    assertFalse(str2.contains("zealously"))
+  }
+
+
+  @Test def testPlagiarismFinderSentenceNotPlagiate() {
+    val input = raw"Jeremy Braddock, Stephen Hock Jeremy Braddock, Stephen Hock (Hrsg.): Directed by Allen Smithee. Foreword by Andrew Sarris."
+    val input2 = raw"Die hatte den mit laut Börsenwerte. Oliver verwendet erfreuten drohte mit, und vermittelte Frau können war geben. Sie Mann Ausdruck Frau sie suspendiert, sah nicht an dem Gemeinde Armenhaus. Oh verhaftet Software erst ihr, das alles er bietet und. Und erfahrungsgemäß Fenster ein schwache noch Andeutungen eigentlich Arzt Stadt Annehmlichkeiten. Der niemals wurden müssen durch Verfahren, man mitlachen sie berühmtes sie Auftrag der sicher Personen, die sozialen Gut damit können Ich durch. Besonders Begriff stand von hatte einigen vor, wenn auf zu Namens offenbar brachte ungewöhnlichen Wächtern betrachtete, tatsächlich wäe und Ausstrecken greisenhafter hierbleiben erklärte herrschte Weise bekämen. Besonders aber die."
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:67]: str1 obsolete. str2=matches.toString

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    val str2 = str1.toString
+    assertEquals(str2, "()")
+    if (str2.contains("Namens offenbar brachte ungewöhnlichen Wächtern")) {
+      assert(false)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:71]: bool containsExpectedPlagiarismText= <condition>
assert(containsExpectedPlagiarismText)

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+    val input2 = raw"Gewinnung Darstellung Da Uranerzen nur wenig Actinium vorhanden spielt diese Quelle keine Rolle für Gewinnung."
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
+    val str2 = str1.toString
+    assertFalse(str2.isEmpty)
+    if (str2.contains("der, scandiumgruppe, das, element, ist, ein")) {
+      assert(true)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:83]: bool containsExpectedPlagiarismText=<condition>
assert(containsExpectedPlagiarismText)

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
...nsistenzbegriffe zusammenfallen: In der klassischen Logik lässt sich aus einem Widerspruch jeder beliebige Satz herleiten (dieser Sachverhalt wird Ex falso quodlibet genannt), d.\xa0h. wenn ein klassischer Kalkül auch nur einen Widerspruch herleiten könnte, also im ersten Sinn inkonsistent wäre, dann könnte er jede Aussage herleiten, wäre also im zweiten Sinn inkonsistent. Wenn umgekehrt ein Kalkül inkonsistent im zweiten Sinn ist, also in ihm jede Aussage herleitbar ist, dann ist insbesondere auch jeder Widerspruch herleitbar und ist er auch inkonsistent im ersten Sinn.\n\n; Korrektheit\n: Ein Kalkül heißt genau dann korrekt (semantisch korrekt), wenn in ihm nur solche Formeln hergeleitet werden können, die auch semantisch gültig sind. Für die klassische Aussagenlogik bedeutet das einfacher: Ein Kalkül ist genau dann korrekt, wenn in ihm nur Tautologien bewiesen und nur gültige Argumente hergeleitet werden können.\n: Ist es in einem aussagenlogischen Kalkül möglich, mindestens ein ungültiges Argument herzuleiten oder mindestens eine Formel zu beweisen, die keine Tautologie ist, dann ist der Kalkül inkorrekt.\n\n; Vollständigkeit\n: Vollständig (semantisch vollständig) heißt ein Ka
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: bool containsExpectedPlagiarismText= str2.contains("maximiano, campos, 1941") && str2.contains("akademie, verlag, berlin")
assert(containsExpectedPlagiarismText)

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: bool containsExpectedPlagiarismText= <condition>
assert(containsExpectedPlagiarismText)

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    val input2 = raw""
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
+    val str2 = str1.toString
+    assert(true)
+    assertTrue(str2.contains("()"))
+    assertFalse(str2.isEmpty)
+    if (str2.contains("natriumdihydrogenphosphat, nah, 2, po, 4, zu, einer")) {
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:96]: bool containsExpectedPlagiarismText=[condition]
assert(containsExpectedPlagiarismText)

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+    val input2 = raw""
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
+    val str2 = str1.toString
+    assert(true)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:93]: obsolete

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+    assertTrue(str2.contains("()"))
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:94]: combine all 3 asserts into one bool?

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+    val input2 = raw""
+    val matches = finder.findPlagiarisms(input + input2, new HyperParameters())
+    val str1 = matches.foreach(println)
```
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala:91]: str1 obsolete. str2=matches.toString

File: plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala (hunk 8 above)
Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: str1 obsolete. str2=matches.toString

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: bool containsExpectedPlagiarismText=[condition]
assert(containsExpectedPlagiarismText)

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: assert( str2.isEmpty)

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: str1 obsolete. str2=matches.toString

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: str1 obsolete. str2=matches.toString

Comment (KarakoA) [inline plagiarismFinder/src/test/scala/de/htwberlin/f4/wikiplag/plagiarism/MatchTest.scala]: combine with previous assert?

**Pull request closed without merging**